localstack-core/localstack/services/sqs/provider.py

```diff
@@ -1232,8 +1233,11 @@ def receive_message(
 
         num = max_number_of_messages or 1
```

Comment:
how about moving the `num` assignment here from the condition block later? this could improve the structure and also allow us to retrieve all messages (with num=-1) using the header

```suggestion
        # override receieve count with value from custom header
        if override := extract_message_count_from_headers(context):
            num = override
        else:
            num = max_number_of_messages or 1
```

Reply:
The current logic allow us to only validate the parameter if no override (or -1) is provided.

I agree we should look into using the override header for this `-1` case but will rather do it in a follow-up PR.